src/gui/sharemanager.cpp

```diff
@@ -401,6 +469,34 @@ void ShareManager::slotSharesFetched(const QJsonDocument &reply)
     emit sharesFetched(shares);
 }
 
+QSharedPointer<UserGroupShare> ShareManager::parseUserGroupShare(const QJsonObject &data)
+{
+    QSharedPointer<Sharee> sharee(new Sharee(data.value("share_with").toString(),
+        data.value("share_with_displayname").toString(),
+        static_cast<Sharee::Type>(data.value("share_type").toInt())));
+
+    QDate expireDate;
+    if (data.value("expiration").isString()) {
+        expireDate = QDate::fromString(data.value("expiration").toString(), "yyyy-MM-dd 00:00:00");
```

Comment:
@FlexW Similar to the other place. Checking `expireDate.isValid()` wouldn't harm. Unless there is a guarantee that `data["expiration"]` is always set and is always in a proper format?

Reply:
See my other comment. The logic depends on invalid dates. We could use something like std::optional to make it clearer?